src/useImageDimensions.ts

```diff
@@ -0,0 +1,61 @@
+import {useEffect, useState} from 'react'
+import {Image, ImageRequireSource} from 'react-native'
+
+export interface URISource {
+  uri: string
+}
+
+/**
+ * @param source either a remote URL or a local file resource.
+ * @returns original image dimensions (width, height and aspect ratio).
+ */
+function useImageDimensions(source: ImageRequireSource | URISource) {
+  const [[dimensions, error], setState] = useState<
+    [{width: number; height: number}?, Error?]
+  >([])
+
+  useEffect(() => {
+    try {
+      if (typeof source === 'number') {
+        const {width, height} = Image.resolveAssetSource(source)
+        setState([{width, height}])
+      } else if (typeof source === 'object' && source.uri) {
+        setState([])
+        Image.getSize(
+          source.uri,
+          (width, height) => setState([{width, height}]),
+          e => setState([dimensions, e]),
```

Comment:
`dimensions` here will refer to the previous dimensions, so if you change the source to one that will give an error, the dimensions will temporarily be cleared, and then when the error happens to dimensions will be set back.

This would be solved by using two `useState`s as well...

Reply:
It is the feature to not ruin the previously built markup in the case when an error happens. You can still handle the error any way you want.

Reply:
But if the source have changed I really don't think that we should return the dimensions of the previous image. That could potentially break the markup that someone is using.